File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
    if (!root) return [];
    let q = [root];
    let res = [];
    while (q.length) {
        let cllength = q.length; // current-level length
        let cla = []; //current level array
        let count = 0;
        while (count < cllength) {
            let cn = q.shift();
            if (cn.left) q.push(cn.left);
            if (cn.right) q.push(cn.right);
            count++;
            cla.push(cn.val);
        }
        res.push(cla);
    }
    return res;
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        finalRes = []
        

        def bfsSol(root):
            queue = deque()
            queue.append(root)

            while queue:
                clLength = len(queue) #current - level length
                cla = [] # current-level array

                for i in range(0, clLength):
                    x = queue.popleft()
                    cla.append(x.val)
                    if x.left:
                        queue.append(x.left)
                    if x.right:
                        queue.append(x.right)
                finalRes.append(cla)

        bfsSol(root)

        return finalRes


            

       


        